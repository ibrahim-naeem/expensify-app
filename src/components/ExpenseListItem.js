
import React from "react";
import { Link } from "react-router-dom";

export const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>Description : {description}</h3>  
    </Link>
    <p>
      Amount : {amount} - CreatedAt : {createdAt}
    </p>
    
  </div>
);
export default ExpenseListItem;