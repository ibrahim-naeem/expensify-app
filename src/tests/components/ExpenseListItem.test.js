
import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import toJson from 'enzyme-to-json';
import '../setupTests';

import { ExpenseListItem } from '../../components/ExpenseListItem';


test('should render expenser list item expense', () => {
    
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
})