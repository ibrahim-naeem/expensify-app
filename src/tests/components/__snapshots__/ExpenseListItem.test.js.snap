// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenser list item expense 1`] = `
<div>
  <[object Object]
    to="/edit/1"
  >
    <h3>
      Description : 
      Gym
    </h3>
  </[object Object]>
  <p>
    Amount : 
    195
     - CreatedAt : 
    0
  </p>
</div>
`;
